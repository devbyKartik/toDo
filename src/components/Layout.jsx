import React, { Component } from 'react';

import ToDo from "./pages/toDo/ToDo";
class Layout extends Component {
  constructor(props) {
    super(props);
    this.state = {
      num: 1,
      btnStatus: false
    }
  }
  handleChange = e => {
    this.setState({
      num: e.target.value
    });
  };
  
  render() {
    const { num, btnStatus } = this.state;
    return (
      <>
         <div className="container-fluid">
          <p>{num}</p>
          <input
            onChange={this.handleChange}
            type="number"
            value={num}
            placeholder="enter id here"
          />
          <button className="btn btn-info" onClick={() => { this.setState({ num: 5 }) }}>Ok!</button>
          <h1>Snapshort Testing</h1>
          <a
            className={btnStatus ? "btn btn-info" : "btn btn-success"}
            onClick={() => { this.setState({ btnStatus: !btnStatus }) }}
          > class change </a>

          <h2 className="hello1">hello</h2>
          
        </div>
        <ToDo />
      
      </>
    )
  }
}

export default Layout;
