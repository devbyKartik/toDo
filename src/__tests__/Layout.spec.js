import React from 'react';
import { configure , shallow} from 'enzyme';
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';
import Layout from "../components/Layout";
configure({ adapter: new Adapter() });

// test for  input type change
it("Updates the state", () => {
    const wrapper = shallow(<Layout/>);
    const input = wrapper.find("input");
    input.simulate("change", { target: { value: 6 } });  // 'value' instead of 'num'
    expect(input.props().value).toEqual(1);
    const p = wrapper.find("p");
    expect(p.props()).toEqual({ children: 6 });
});


// test for change state using button
describe('The Count Up Button', () => {
  it('increments state count by 1 on click', () => {
    const wrapper = shallow(<Layout/>);
    wrapper.find('button').props().onClick();
    expect(wrapper.state().num).toEqual(5); // 'value' instead of 'num'
    const input = wrapper.find("input");
    expect(input.props().value).toEqual(5);
  });
});

 

describe('snapshort', () => {
  it('code for test using snapshort', () => {
    const tree = renderer.create(<h1>Snapshort Testing</h1>).toJSON()
    expect(tree).toMatchSnapshot()
  });
});


// test for check className on button change
describe('The Count Up Button', () => {
  it('increments state count by 1 on click', () => {
    const wrapper = shallow(<Layout/>);
    wrapper.find('a').props().onClick();
    const a=wrapper.find('a');
    expect(a.hasClass('btn btn-info')).toEqual(true);
  });
});


// test for get single h2 tag data
describe('The Count Up Button', () => {
  it('increments state count by 1 on click', () => {
    const wrapper = shallow(<Layout/>);
    const a=wrapper.find('.hello1');
  });
});




 