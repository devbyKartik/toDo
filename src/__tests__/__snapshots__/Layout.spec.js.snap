// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshort code for test using snapshort 1`] = `
<h1>
  Snapshort Testing
</h1>
`;

